src/cpp/session/modules/SessionFiles.cpp

```diff
@@ -1070,10 +1102,13 @@ bool handleFileUploadRequestAsync(const http::Request& request,
    uploadTokenJson[kUploadFilename] = pUploadState->fileName;
    uploadTokenJson[kUploadedTempFile] = pUploadState->tmpFile.getAbsolutePath();
    uploadTokenJson[kUploadTargetDirectory] = destDir.getAbsolutePath();
+   uploadTokenJson[kUnzipFound] = unzipFound; // Include here to be used by completeUpload()
```

Comment:
I have unzip found in two places because both GWT and completeUpload() need access to the flag, and I couldn't see an obvious way to access the flag in the token in GWT.

Reply:
I'm probably missing something, but couldn't you add a JSNI accessor to the FileUploadToken class (similar to https://github.com/rstudio/rstudio/pull/9231/files#diff-484e33a9fc038459f5594561f5a619c7ce46dc66e8cedc788d32e0c5b89c9b39R35-R37)?

Reply:
That was the first thing I tried, but for whatever reason copying the exact same JSNI accessors into FileUploadToken causes a reference error:
<img width="368" alt="Screen Shot 2021-04-14 at 11 54 34 AM" src="https://user-images.githubusercontent.com/56326543/114763909-30e4a980-9d18-11eb-9d81-4b9399980cc1.png">

Using the same code in the PendingFileUpload object does not cause that error, and I couldn't figure out why. I'm likely making some newbie mistake here. 

Here's snippets of what I tried:
FileUploadToken.java:
```
public class FileUploadToken extends JavaScriptObject
{
   protected FileUploadToken()
   {
   }

   public final native Boolean getUnzipFound() /*-{
      return this.unzipFound;
   }-*/;

   public final native Boolean getIsZip() /*-{
      return this.isZip;
   }-*/;
}
```

FileUpload.java:
```
public void execute(PendingFileUpload pendingUpload)
            {
               FileUploadToken token = pendingUpload.getToken();
               Boolean unzipFound = token.getUnzipFound();             <=== Reference Error
               ...
```

Reply:
Weird! That looks correct to me. In some cases, rebuilding GWT from scratch with `ant clean; ant devmode` resolves the issue. Is that worth a shot? If it doesn't help, then I think leaving things as-is is fine -- we occasionally see inexplicable GWT errors like this that we usually, unfortunately, lack the time to fully diagnose or understand...

Reply:
No luck with cleaning/rebuilding, but I did catch a break by adding in getters for all the members of the JSON object, not just the two that I needed. Now it works great! Definitely not the behavior I expected